brainstorm/src/components/items.tsx

```diff
@@ -78,15 +91,28 @@ export function ItemsView(props: {
 	}
 }
 
-/**
- * Removes a node from its parent {@link Items}.
- * If the note is not in an {@link Items}, it is left unchanged.
- */
-export function deleteItem(item: Item): void {
-	const parent = Tree.parent(item);
-	// Use type narrowing to ensure that parent is Items as expected for an Item.
-	if (Tree.is(parent, Items)) {
-		const index = parent.indexOf(item);
-		parent.removeAt(index);
+export interface ItemParent {
+	/**
+	 * Removes a child node.
+	 * If the node is not in an {@link Items}, it is left unchanged.
+	 */
+	deleteItem(item: Item): void;
+}
+
+interface HasItemParent {
+	get [ItemParentSymbol](): ItemParent;
+}
+
+export function removeItemFromParent(item: Item): void {
+	const parent = Tree.parent(item) as Partial<HasItemParent>;
```

Comment:
the as cast should be inside the `if (parent !== undefined) {` so we don't cast away `undefined` here.